panel/src/components/Forms/FormButtons.vue

```diff
@@ -22,29 +58,174 @@
 </template>
 
 <script>
+import dayjs from "dayjs";
+import relativeTime from 'dayjs/plugin/relativeTime';
+dayjs.extend(relativeTime);
+
 export default {
+  data() {
+    return {
+      now: dayjs(),
+      heartbeat: null
+    }
+  },
   computed: {
+    canUnlock() {
+      return this.lock.time + (60 * 2) < this.now.unix();
+    },
     hasChanges() {
       return this.$store.getters["form/hasChanges"](this.id);
     },
+    hasLock() {
+      return this.lock !== null;
+    },
+    hasUnlock() {
+      return this.unlock !== null;
+    },
     id() {
-      return this.$store.state.form.current;
+      return this.$store.getters["form/current"];
+    },
+    isDisabled() {
+      return this.$store.getters["form/isDisabled"];
+    },
+    lock() {
+      return this.$store.getters["form/lock"];
+    },
+    lastLocked() {
+      return this.now.to(dayjs(this.lock.time * 1000));
+    },
+    unlock() {
+      return this.$store.getters["form/unlock"];
     },
-    isLocked() {
-      return this.$store.getters["form/isLocked"];
+    unlockMsg() {
+      return this.canUnlock ? this.$t('unlock') : this.$t('unlock.disabled');
+    },
+    user() {
+      return this.$store.state.user.current.id;
+    }
+  },
+  watch: {
+    hasChanges(current, previous) {
+      // if user started to make changes,
+      // start setting lock on each heartbeat
+      if (current === true && previous == false) {
+        this.onHeartbeat(this.setLock, 30);
+        console.log('Has changes => setting lock...');
+
+      // if user reversed changes manually,
+      // remove lock and listen to lock from other users again
+      } else if (current === false && previous == true) {
+        this.unsetLock();
+        console.log('No changes anymore => unset lock and listening...');
+      }
+    },
+    id() {
+      // whenever the model id changes, make sure to reset
+      // heartbeat listener again
+      if (this.hasChanges === false) {
+        this.onHeartbeat(this.listenToLock);
+        console.log('start listening...');
+      }
     }
   },
   created() {
     this.$events.$on("keydown.cmd.s", this.save);
+    setInterval(() => {
+      this.now = dayjs();
+    }, 1000);
   },
   destroyed() {
     this.$events.$off("keydown.cmd.s", this.save);
   },
   methods: {
-    reset() {
+    listenToLock() {
+      // call API to check for lock
+      return this.$api.get(this.$route.path + "/lock").then(response => {
+
+        // if a lock is set and it's not from the current user
+        // dispatch lock information to store
+        if (response.lock === true && response.user !== this.user) {
```

Comment:
What happens if the user locks an object and then tries to access the same object with the same account, but from a different device? Won't it appear unlocked there even though the local storage is only on the other device?

Reply:
Yes, but honestly I feel like preventing this might become veeery complex. So I left that agency with the user.

Reply:
What we could do: If the lock is set to the current user but there's nothing in the local storage on the current device, treat the lock like a lock of a different user. The message would be something like "this page/file is locked by you on a different device".

Reply:
@lukasbestle with the current implementation we can't do it at the moment, since the backend returns no lock for the current user if the lock is by himself/herself.